natiami/assignment1/Q1.cpp

```diff
@@ -0,0 +1,48 @@
+#include<iostream>
+#include<string>
+#include<vector>
+#include<fstream>
+#include <algorithm>
+#include <array> 
+
+using namespace std;
+
+bool checkAnagrams(string str1, string str2, bool caseSensitive){
+	vector<int> letters(52, 0); // array for letters
+
+	if (str1.length() != str2.length()) return false; // if size differs return false right away
+
+	for (size_t i = 0; i < str1.length(); i++)
+	{
+		letters[int(str1[i]) - 65]++;
+		letters[int(str2[i]) - 65]--;
+	}
+
+	// If algorithm is not case sensitive
+	if (!caseSensitive){
+		for (size_t i = 0; i < letters.size()/2; i++)
+			if (letters[i] != 0 && letters[i + 32] == 0) return false;

```

Comment:
Sorry but I did not understand what this line is supposed to check.
Consider the following input (not anagrams):
`AaBbb`
`AaaBb`
For letters A and B, `letters[i+32] ` is not zero;
for every other letter, `letters[i]` is zero;
so this function returns `true` for this input.

Please fix this code, or clarify if I am missing something here :)